execute.go

```diff
@@ -112,17 +118,23 @@ func killProcessOnTimeout(cmd *exec.Cmd, timeout time.Duration) (int, error) {
 	return exitCode, nil
 }
 
-// gets the exit code from error
+func processExitCode(err error) int {
+	var exitCode int
+	if err != nil {
+		var exitErr error
+		if exitCode, exitErr = getExitCode(err); exitErr != nil {
+			exitCode = 127
```

Comment:
This need more context instead of returning 127. I will make some changes after getting this kill process thing.